packages/canner-storage/src/ImgurStorage.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+
+import axios from 'axios';
+import {getUploadPercent} from './utils';
+import type {UploadOptions, OnProgressType} from './types';
+
+const IMGUR_IMAGE_API = 'https://api.imgur.com/3/image/';
+const MASHAPE_API = 'https://imgur-apiv3.p.mashape.com/3/image';
+
+export default class ImgurStorage {
+  clientId: string;
+  mashapeKey: string;
+  
+  constructor({clientId = '', mashapeKey = ''}: {clientId: string, mashapeKey?: string}) {
+    this.clientId = clientId;
+    this.mashapeKey = mashapeKey;
+  }
+
+  getUploadUrl() {
+    return {
+      url: this.mashapeKey ? MASHAPE_API : IMGUR_IMAGE_API
+    };
+  }
+
+  upload(file: File, options: UploadOptions, onProgress: OnProgressType) {
+    let UPLOAD_API = IMGUR_IMAGE_API;
```

Comment:
I think we can get the `UPDATE_API` with calling `this.getUploadUrl()` here.

Reply:
Good point. I've changed it. Thanks!